src/Microsoft.AspNetCore.TestHost/HostBuilderTestServerExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.TestHost
+{
+    public static class HostBuilderTestServerExtensions
+    {
+        /// <summary>
+        /// Builds and Starts the Host using a TestServer.
+        /// </summary>
+        /// <param name="hostBuilder"></param>
+        /// <param name="featureCollection"></param>
+        /// <returns></returns>
+        public static async Task<TestServer> StartTestServerAsync(this IHostBuilder hostBuilder, IFeatureCollection featureCollection = null)
```

Comment:
I think we should stop making extension methods on the builder that starts. It's been a pattern that I've hardly seen anyone use and it breaks the builder flow and the host flow. How do you dispose the host in this case? It's lost...

Reply:
Agreed. I think it's better to just have TestServer.StartAsync().